commons-geometry-euclidean/src/main/java/org/apache/commons/geometry/euclidean/threed/Plane.java

```diff
@@ -16,174 +16,193 @@
-     * shared (except for immutable objects).</p>
-     * @return a new hyperplane, copy of the instance
+     * @return a new plane
+     * @throws IllegalNormException if the norm of the given values is zero, NaN, or infinite.
      */
-    @Override
-    public Plane copySelf() {
-        return new Plane(this);
+    public static Plane fromNormal(final Vector3D normal, final DoublePrecisionContext precision){
+        return fromPointAndNormal(Vector3D.ZERO, normal, precision);
     }
 
-    /** Reset the instance as if built from a point and a normal.
-     * @param p point belonging to the plane
-     * @param normal normal direction to the plane
-     * @exception IllegalArgumentException if the normal norm is too small
+    /**
+     * Build a plane from a point and a normal.
+     * 
+     * @param p         point belonging to the plane
+     * @param normal    normal direction to the plane
+     * @param precision precision context used to compare floating point values
+     * @return a new plane
+     * @throws IllegalNormException if the norm of the given values is zero, NaN, or infinite.
      */
-    public void reset(final Vector3D p, final Vector3D normal) {
-        setNormal(normal);
-        originOffset = -p.dot(w);
-        setFrame();
+    public static Plane fromPointAndNormal(final Vector3D p, final Vector3D normal, final DoublePrecisionContext precision) {
+        Vector3D w = normal.normalize();
+        double originOffset = -p.dot(w);
+        Vector3D projectedOrigin = calculateOrigin(w, originOffset);
+        Vector3D u = w.orthogonal();
+        Vector3D v = w.cross(u);
+        return new Plane(u, v, w, projectedOrigin, originOffset, precision);
     }
-
-    /** Reset the instance from another one.
-     * <p>The updated instance is completely independant of the original
-     * one. A deep reset is used none of the underlying object is
-     * shared.</p>
-     * @param original plane to reset from
+    
+    /**
+     * Build a plane from three points.
+     * <p>
+     * The plane is oriented in the direction of {@code (p2-p1) ^ (p3-p1)}
+     * </p>
+     * 
+     * @param p1        first point belonging to the plane
+     * @param p2        second point belonging to the plane
+     * @param p3        third point belonging to the plane
+     * @param precision precision context used to compare floating point values
+     * @return a new plane
+     * @throws IllegalNormException if the points do not constitute a plane
      */
-    public void reset(final Plane original) {
-        originOffset = original.originOffset;
-        origin       = original.origin;
-        u            = original.u;
-        v            = original.v;
-        w            = original.w;
+    public static Plane fromPoints(final Vector3D p1, final Vector3D p2, final Vector3D p3,
+            final DoublePrecisionContext precision) {
+        return Plane.fromPointAndNormal(p1, p2.subtract(p1).cross(p3.subtract(p1)), precision);
```

Comment:
Rather than calling `fromPointAndNormal()` to construct the Plane here, why not call `fromPointandPlaneVectors`? That way, the `u` vector will be oriented to point from `p1` to `p2`, giving the caller more control over the orientation of the plane by his/her choice of points.

Reply:
I'm struggeling with this one. Some PolyhedronsSetTest fail, when I make the change ... I have to investigate this

Reply:
done